libs/common/RespReadUtils.cs

```diff
@@ -241,19 +473,32 @@ public static bool ReadByteArrayWithLengthHeader(out byte[] result, ref byte* pt
         /// </summary>
         public static bool ReadBoolWithLengthHeader(out bool result, ref byte* ptr, byte* end)
         {
-            //$1\r\n1\r\n
-            //$1\r\n0\r\n
             result = false;
-            if (ptr + 7 >= end)
+
```

Comment:
suggestion: can this be fast-pathed in the "we have 8+ bytes" case using pre-defined `ulong`, masking off one byte?

Reply:
That's a good idea! We are doing something similar for fast-parsing array headers of common commands. At a minimum we can bypass the checks for the length header by just checking for '$1\r\n'.

Reply:
I have added a fast path for the common case '**$1\r\n**'. Performance improvements are definitely visible.

<details>
<summary>Performance impact</summary><p>

| Method         | Job    | EnvironmentVariables | Runtime  | testCase    | Mean     | Error     | StdDev    |
|--------------- |------- |--------------------- |--------- |------------ |---------:|----------:|----------:|
| ReadBoolHeader (old) | .NET 6 | Empty                | .NET 6.0 | $1\r\n0\r\n | 7.982 ns | 0.0343 ns | 0.0321 ns |
| ReadBoolHeader (old) | .NET 8 | DOTNET_TieredPGO=0   | .NET 8.0 | $1\r\n0\r\n | 6.795 ns | 0.0017 ns | 0.0013 ns |
| ReadBoolHeader (old) | .NET 6 | Empty                | .NET 6.0 | $1\r\n1\r\n | 7.954 ns | 0.0279 ns | 0.0261 ns |
| ReadBoolHeader (old) | .NET 8 | DOTNET_TieredPGO=0   | .NET 8.0 | $1\r\n1\r\n | 6.797 ns | 0.0034 ns | 0.0032 ns |
| ReadBoolHeader (new) | .NET 6 | Empty                | .NET 6.0 | $1\r\n0\r\n | 7.085 ns | 0.0088 ns | 0.0073 ns |
| ReadBoolHeader (new)| .NET 8 | DOTNET_TieredPGO=0   | .NET 8.0 | $1\r\n0\r\n | 4.156 ns | 0.0018 ns | 0.0017 ns |
| ReadBoolHeader (new)| .NET 6 | Empty                | .NET 6.0 | $1\r\n1\r\n | 7.082 ns | 0.0046 ns | 0.0041 ns |
| ReadBoolHeader (new)| .NET 8 | DOTNET_TieredPGO=0   | .NET 8.0 | $1\r\n1\r\n | 4.163 ns | 0.0059 ns | 0.0050 ns |

</p>
</details>